platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+import org.opensaml.xml.XMLObject;
+import org.opensaml.xml.security.credential.UsageType;
+import org.opensaml.xml.validation.ValidationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+
+import ddf.security.PropertiesLoader;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.encryption.EncryptionService;
+import ddf.security.samlp.MetadataConfigurationParser;
+import ddf.security.samlp.SamlProtocol;
+import ddf.security.samlp.SimpleSign;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+@Path("/")
+public class IdpEndpoint implements Idp {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IdpEndpoint.class);
+
+    private static final String IDP_STATE_OBJ = "IDP_STATE_OBJ";
+
+    private static final String PKI = "pki";
+
+    private static final String GUEST = "guest";
+
+    private static final String USER_PASS = "up";
+
+    private static final String SAML_RESPONSE = "SAMLResponse";
+
+    private static final String ACS_URL = "ACSURL";
+
+    private static final String HTTP_POST_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+
+    private static final String SAML_SOAP_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:SOAP";
+
+    private static final String PAOS_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:PAOS";
+
+    private static final String HTTP_REDIRECT_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect";
+
+    private static final String HTTP_ARTIFACT_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact";
+
+    private PKIAuthenticationTokenFactory tokenFactory;
+
+    private SecurityManager securityManager;
+
+    private Crypto signatureCrypto;
+
+    private Crypto encryptionCrypto;
+
+    private EncryptionService encryptionService;
+
+    private String signaturePropertiesPath;
+
+    private String encryptionPropertiesPath;
+
+    private Map<String, EntityDescriptor> serviceProviders = new HashMap<>();
+
+    private String indexHtml;
+
+    private String submitForm;
+
+    private String redirectPage;
+
+    private Boolean strictSignature = true;
+
+    public void init() {
+        try {
+            indexHtml = IOUtils.toString(IdpEndpoint.class.getResourceAsStream("/html/index.html"));
+            submitForm = IOUtils.toString(
+                    IdpEndpoint.class.getResourceAsStream("/templates/submitForm.handlebars"));
+            redirectPage = IOUtils.toString(
+                    IdpEndpoint.class.getResourceAsStream("/templates/redirect.handlebars"));
+        } catch (Exception e) {
+            LOGGER.error("Unable to load index page for IDP.", e);
+        }
+    }
+
+    private void parseSpMetadata(List<String> spMetadata) {
+        if (spMetadata != null) {
+            try {
+                serviceProviders = MetadataConfigurationParser.buildEntityDescriptors(spMetadata);
```

Comment:
Does this have to be a static method? This class is already huge and will be a nightmare to unit test already so if we can at least avoid static methods that'd be great,